central/resourcecollection/datastore/datastore_impl.go

```diff
@@ -203,26 +379,48 @@ func (ds *datastoreImpl) GetBatch(ctx context.Context, ids []string) ([]*storage
 }
 
 func (ds *datastoreImpl) AddCollection(ctx context.Context, collection *storage.ResourceCollection) error {
```

Comment:
nit : Since it sets the Id too, would it be better for the func to return the id as well?
 
Yes, it sets it on the pointer to obj and it would be updated for the caller too, but the caller is ideally not looking at this implementation before calling `AddCollection` and just looking at datastore interface. Another way could be to add a comment on the interface that says id would be set in place.

Reply:
Since the collection passed into the function passes by reference the Id field just gets set on the object. The way it's currently been refactored is that the Id will only get set upon the operation being successful, I'll go ahead and add some unit tests to ensure we follow that expected functionality.

I'll be sure to add a note on the interface for that. As far as whether we should also return Id I'm unsure, I'd think it's not necessary since we should be returning nil as the object in the service layer upon error anyways.